java/ql/src/semmle/code/java/dataflow/internal/rangeanalysis/ModulusAnalysisSpecific.qll

```diff
@@ -0,0 +1,119 @@
+module Private {
+  private import java as J
+  private import semmle.code.java.dataflow.SSA as Ssa
+  private import semmle.code.java.dataflow.RangeUtils as RU
+  private import semmle.code.java.controlflow.Guards as G
+  private import semmle.code.java.controlflow.BasicBlocks as BB
+
+  class BasicBlock = BB::BasicBlock;
+
+  class SsaVariable = Ssa::SsaVariable;
+
+  class SsaPhiNode = Ssa::SsaPhiNode;
+
+  class Expr = J::Expr;
+
+  class Guard = G::Guard;
+
+  class ConstantIntegerExpr = RU::ConstantIntegerExpr;
+
+  class ConditionalExpr = J::ConditionalExpr;
+
+  /** An addition or an assign-add expression. */
+  class AddExpr extends J::Expr {
+    AddExpr() { this instanceof J::AddExpr or this instanceof J::AssignAddExpr }
+
+    /** Gets the LHS operand of this add expression. */
+    Expr getLhs() {
+      result = this.(J::AddExpr).getLeftOperand()
+      or
+      result = this.(J::AssignAddExpr).getDest()
+    }
+
+    /** Gets the RHS operand of this add expression. */
+    Expr getRhs() {
+      result = this.(J::AddExpr).getRightOperand()
+      or
+      result = this.(J::AssignAddExpr).getRhs()
+    }
+  }
+
+  /** Represent a subtraction or an assign-sub expression. */
```

Comment:
```suggestion
  /** A subtraction or an assign-sub expression. */
```

Reply:
Hahh, good point. The assignment was not the only one...

Reply:
I pushed the fix for the comments.